crates/guest-rust/macro/src/lib.rs

```diff
@@ -168,31 +178,40 @@ fn parse_source(
     let mut resolve = Resolve::default();
     resolve.features.extend(features.iter().cloned());
     let mut files = Vec::new();
+    let mut pkgs = Vec::new();
     let root = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
-    let mut parse = |path: &Path| -> anyhow::Result<_> {
-        // Try to normalize the path to make the error message more understandable when
-        // the path is not correct. Fallback to the original path if normalization fails
-        // (probably return an error somewhere else).
-        let normalized_path = match std::fs::canonicalize(path) {
-            Ok(p) => p,
-            Err(_) => path.to_path_buf(),
-        };
-        let (pkg, sources) = resolve.push_path(normalized_path)?;
-        files.extend(sources);
-        Ok(pkg)
+    let mut parse = |paths: &Vec<String>| -> anyhow::Result<()> {
```

Comment:
Can this be `&[PathBuf]` instead of `&Vec<String>`? That keeps the type-level annotation that these are paths and additionally avoids the dance of `into_os_string().into_string().unwrap()` below too